⚠️  Add --raw flag for clusterctl generate provider subcommand

aayushrangwala opened the pull request:
**What this PR does / why we need it**:
Add --raw flag for clusterctl generate provider subcommand  for unprocessed yaml

  - Added a New flag --raw of type bool in the provider subcommand 
  - If flag provided, the variables processing will be skipped while generating provider yaml

Signed-off-by: Ayush Rangwala <arangwala@vmware.com>
 
Since we are in the process to deprecate `clusterctl config provider --infrastructure aws` commands and migrating to `clusterctl generate provider`, based on [this](https://github.com/kubernetes-sigs/cluster-api/issues/3360#issuecomment-662677592) discussion, we decided to have a `--raw` flag to give the same functionality as of the `config` subcommand

Fixes #4650

enxebre commented:
/ok-to-test

fabriziopandini commented on cmd/clusterctl/client/repository/components.go:
```diff
@@ -189,7 +189,7 @@ func NewComponents(input ComponentsInput) (Components, error) {
 
 	processedYaml := input.RawYaml
-	if !input.Options.SkipVariables {
+	if !input.Options.SkipVariableProcess {
```
Should this be SkipComponentsTemplateProcess (or simply SkipTemplateProcess)?
Also, should we return here when skipping template process, given that all the following code assumes we have a valid yaml, which it is not the case for RawYaml

fabriziopandini reviewed (commented):
@aayushrangwala thanks for your first PR in Cluster API 🎉 🥳 !!!!

changes sgmt apart a small nit about flag naming.
Also, would you mind to open an issue so we can create a separated PR and rename https://github.com/kubernetes-sigs/cluster-api/blob/df2ab2f54e863124dde1fddca19694d3ab8190b1/cmd/clusterctl/client/repository/template.go#L89 in the same way for sake of consistency

fabriziopandini commented:
Also the PR should be marked as ⚠️ because we are changing a fields name (let's add to the title "; rename SkipVariables into SkipTemplateProcess")

fabriziopandini commented:
I'm picking up this work so we get this into the release

aayushrangwala commented:
> I'm picking up this work so we get this into the release

Okay @fabriziopandini. I already started addressing review comments and was anyway going to wrap up in a day or 2. But if its on a very high priority then you can pick up

fabriziopandini commented:
/close
in favor of https://github.com/kubernetes-sigs/cluster-api/pull/4768

@aayushrangwala thanks for your amazing work on this PR; the work was really near the end line, and this is why we decided to make it into the release. I made sure to give you credits for the commit and to make it visible your help.

If you are interested to continue this work, you can pick up 

> Also, would you mind to open an issue so we can create a separated PR and rename https://github.com/kubernetes-sigs/cluster-api/blob/df2ab2f54e863124dde1fddca19694d3ab8190b1/cmd/clusterctl/client/repository/template.go#L89 in the same way for sake of consistency

aayushrangwala commented:
> /close
> in favor of #4768
> 
> @aayushrangwala thanks for your amazing work on this PR; the work was really near the end line, and this is why we decided to make it into the release. I made sure to give you credits for the commit and to make it visible your contribution.
> 
> If you are interested to continue this work, you can pick up
> 
> > Also, would you mind to open an issue so we can create a separated PR and rename https://github.com/kubernetes-sigs/cluster-api/blob/df2ab2f54e863124dde1fddca19694d3ab8190b1/cmd/clusterctl/client/repository/template.go#L89
> >  in the same way for sake of consistency

@fabriziopandini  Thanks a lot. Sure I will pick it up
